### ISSUE TITLE
关于对比实验的一些请教

### ISSUE THREAD

**Issue by Baccarali** [Henryjiepanli/Uncertainty-aware-Network#5]
非常感谢你们的工作！在您的paper中看到了对比实验中的多个模型，请问在这些模型都是此项目中的相同的配置和超参数下进行的吗？感谢！

**Comment by Henryjiepanli** [comment]
嗯嗯，是的，一些实在复现不了的，我就是直接摘抄论文汇报精度，这一点我在论文中有提及。


-----原始邮件-----
发件人:Baccarali ***@***.***>
发送时间:2024-04-03 11:19:36 (星期三)
收件人: Henryjiepanli/Uncertainty-aware-Network ***@***.***>
抄送: Subscribed ***@***.***>
主题: [Henryjiepanli/Uncertainty-aware-Network] 关于对比实验的一些请教 (Issue #5)






非常感谢你们的工作！在您的paper中看到了对比实验中的多个模型，请问在这些模型都是此项目中的相同的配置和超参数下进行的吗？感谢！

**Comment by Baccarali** [comment]
您好，可以提供heatmap绘制的代码吗

**Comment by Henryjiepanli** [comment]
> 您好，可以提供heatmap绘制的代码吗

def heatmap(feat_viz, ori_img, w, h, save_path=None):
    feat_viz = torch.mean(feat_viz, dim=1, keepdim=True).data.cpu().numpy().squeeze()
    feat_viz = (feat_viz - feat_viz.min()) / (feat_viz.max() - feat_viz.min() + 1e-8)

    ori_img = ori_img.data.cpu().numpy().squeeze()
    ori_img = ori_img.transpose((1, 2, 0))
    ori_img = ori_img * np.array((0.5, 0.5, 0.5)) + np.array((0.5, 0.5, 0.5))
    ori_img = ori_img[:, :, ::-1]
    # img = (img - img.min()) / (img.max() - img.min() + 1e-8)
    ori_img = np.uint8(255 * ori_img)
    feat_viz = np.uint8(255 * feat_viz)
    feat_viz = cv2.applyColorMap(feat_viz, cv2.COLORMAP_JET)
    feat_viz = cv2.resize(feat_viz, (w, h))
    ori_img = cv2.resize(ori_img, (w, h))
    # print(feat_viz.shape, ori_img.shape)
    # feat_viz = cv2.addWeighted(ori_img, 0.3, feat_viz, 0.7, 0)

    cv2.imwrite(save_path, feat_viz) 如果需要叠在原图上，可以取消注释

**Comment by Henryjiepanli** [comment]
> 您好，可以提供heatmap绘制的代码吗

def heatmap(feat_viz, ori_img, w, h, save_path=None):
    feat_viz = torch.mean(feat_viz, dim=1, keepdim=True).data.cpu().numpy().squeeze()
    feat_viz = (feat_viz - feat_viz.min()) / (feat_viz.max() - feat_viz.min() + 1e-8)

    ori_img = ori_img.data.cpu().numpy().squeeze()
    ori_img = ori_img.transpose((1, 2, 0))
    ori_img = ori_img * np.array((0.5, 0.5, 0.5)) + np.array((0.5, 0.5, 0.5))
    ori_img = ori_img[:, :, ::-1]
    # img = (img - img.min()) / (img.max() - img.min() + 1e-8)
    ori_img = np.uint8(255 * ori_img)
    feat_viz = np.uint8(255 * feat_viz)
    feat_viz = cv2.applyColorMap(feat_viz, cv2.COLORMAP_JET)
    feat_viz = cv2.resize(feat_viz, (w, h))
    ori_img = cv2.resize(ori_img, (w, h))
    # print(feat_viz.shape, ori_img.shape)
    # feat_viz = cv2.addWeighted(ori_img, 0.3, feat_viz, 0.7, 0)

    cv2.imwrite(save_path, feat_viz)
